src/ctapipe/calib/camera/calibrator.py

```diff
@@ -282,6 +293,44 @@ def _calibrate_dl1(self, event, tel_id):
         # store the results in the event structure
         event.dl1.tel[tel_id] = dl1
 
+    def _process_calib(self, event, tel_id):
+        waveforms = event.r1.tel[tel_id].waveform
+        if self._check_r1_empty(waveforms):
+            return
+
+        if self.image_extractor_type.tel[tel_id] not in [
+            "LocalPeakWindowSum",
+            "FixedWindowSum",
+        ]:
+            raise ValueError(
+                f"Invalid ImageExtractor '{self.image_extractor_type.tel[tel_id]}'."
+                "Valid options: LocalPeakWindowSum FixedWindowSum."
+            )
+
+        n_channels, n_pixels, n_samples = waveforms.shape
+
+        broken_pixels = event.mon.tel[tel_id].pixel_status.hardware_failing_pixels
+        if broken_pixels is None:
+            broken_pixels = np.zeros(n_pixels, dtype=bool)
+
+        # TODO: See #1836 (process all provided gains in the ImageExtractor)
+        no_gain_selection = np.zeros((n_channels, n_pixels), dtype=np.int64)
+
+        if n_channels == 2:
```

Comment:
should we maybe force to have two channels here since we need both for calibrations? i.e. raising an error if `n_channels != 2` ?

Reply:
But then it will not work with a single gain camera, which is currently supported (I guess)

Reply:
I think we should add an API back to the ImageExtractor for non-gain-selected events instead of special casing two event types here.

See https://github.com/cta-observatory/ctapipe/issues/1836